src/Symfony/Component/String/BinaryString.php

```diff
@@ -0,0 +1,410 @@
+
+    public function beforeLast(string $needle, bool $includeNeedle = false): parent
+    {
+        $str = clone $this;
+
+        if ('' === $needle) {
+            $str->string = '';
+        } else {
+            $i = $this->ignoreCase ? strripos($this->string, $needle) : strrpos($this->string, $needle);
+            $str->string = false === $i ? '' : substr($str->string, 0, $i + ($includeNeedle ? \strlen($needle) : 0));
+        }
+
+        return $str;
+    }
+
+    public function camel(bool $lcFirst): parent
+    {
+        $str = clone $this;
+        $str->string = str_replace(' ', '', ucwords(preg_replace('/[^a-zA-Z0-9\x7f-\xff]++/', ' ', $this->string)));
+
+        if ($lcFirst) {
+            $str->string = lcfirst($str->string);
+        }
+
+        return $str;
+    }
+
+    public function chunk(int $length = 1): array
+    {
+        if (1 > $length) {
+            throw new InvalidArgumentException('The maximum length of each segment must be greater than zero.');
+        }
+
+        if ('' === $this->string) {
+            return [];
+        }
+
+        $chunks = [];
+
+        foreach (str_split($this->string, $length) as $chunk) {
+            $chunks[] = $str = clone $this;
+            $str->string = $chunk;
+        }
+
+        return $chunks;
+    }
+
+    public function endsWith(string $suffix): bool
+    {
+        if ('' === $suffix) {
+            return true;
+        }
+
+        return \strlen($this->string) - \strlen($suffix) === ($this->ignoreCase ? strripos($this->string, $suffix) : strrpos($this->string, $suffix));
+    }
+
+    public function equalsTo(string $string): bool
+    {
+        if ($this->ignoreCase) {
+            return 0 === strcasecmp($string, $this->string);
+        }
+
+        return $string === $this->string;
+    }
+
+    public function folded(): parent
+    {
+        $str = clone $this;
+        $str->string = strtolower($str->string);
+
+        return $str;
+    }
+
+    public function indexOf(string $needle, int $offset = 0): ?int
+    {
+        if ('' === $needle) {
+            return null;
+        }
+
+        $i = $this->ignoreCase ? stripos($this->string, $needle, $offset) : strpos($this->string, $needle, $offset);
+
+        return false === $i ? null : $i;
+    }
+
+    public function indexOfLast(string $needle, int $offset = 0): ?int
+    {
+        if ('' === $needle) {
+            return null;
+        }
+
+        $i = $this->ignoreCase ? strripos($this->string, $needle, $offset) : strrpos($this->string, $needle, $offset);
+
+        return false === $i ? null : $i;
+    }
+
+    public function isUtf8(): bool
+    {
+        return '' === $this->string || preg_match('//u', $this->string);
+    }
+
+    public function join(array $strings): parent
+    {
+        $str = clone $this;
+        $str->string = implode($str->string, $strings);
+
+        return $str;
+    }
+
+    public function length(): int
+    {
+        return \strlen($this->string);
```

Comment:
should it account for mb_ function overloading? which is still possible AFAIK

Reply:
Good question. Answer is no :)
That would penalize everyone for badly configured PHP. They should fix their config instead.

Reply:
agree, but should we block using this class then? i.e. throw in its constructor maybe :/

Reply:
Not our problem... Theirs. If we start doing anything to support broken configuration, where should we stop? Contributor time should not be spent on these.